libfwupdplugin/fu-device.c

```diff
@@ -2524,6 +2663,19 @@ fu_device_sleep_with_progress (FuDevice *self, guint delay_secs)
 	}
 }
 
+static void
+fu_device_ensure_battery_inhibit (FuDevice *self)
+{
+	FuDevicePrivate *priv = GET_PRIVATE (self);
+	if (priv->battery_level == 0)
+		return;
+	if (priv->battery_level >= fu_device_get_battery_threshold (self)) {
+		fu_device_uninhibit (self, "battery");
+		return;
+	}
+	fu_device_inhibit (self, "battery", "Battery level is too low");
```

Comment:
I wonder if you should prefix with the device name to disambiguate from host?

Reply:
I don't think we need to, it's per device rather than per host. The _other_ system one would have a prefix however.